res/res_pjsip/config_transport.c

```diff
@@ -1853,10 +1833,12 @@ int ast_sip_initialize_sorcery_transport(void)
 	ast_sorcery_object_field_register_custom(sorcery, "transport", "require_client_cert", "", transport_tls_bool_handler, require_client_cert_to_str, NULL, 0, 0);
 	ast_sorcery_object_field_register_custom(sorcery, "transport", "allow_wildcard_certs", "", transport_tls_bool_handler, allow_wildcard_certs_to_str, NULL, 0, 0);
 	ast_sorcery_object_field_register_custom(sorcery, "transport", "method", "", transport_tls_method_handler, tls_method_to_str, NULL, 0, 0);
-	ast_sorcery_object_field_register_custom(sorcery, "transport", "tcp_keepalive_enable", "no", transport_tcp_keepalive_bool_handler, NULL, NULL, 0, 0);
-	ast_sorcery_object_field_register_custom(sorcery, "transport", "tcp_keepidle_time", "30", transport_tcp_keepalive_int_handler, NULL, NULL, 0, 0);
-	ast_sorcery_object_field_register_custom(sorcery, "transport", "tcp_keepintvl_time", "1", transport_tcp_keepalive_int_handler, NULL, NULL, 0, 0);
-	ast_sorcery_object_field_register_custom(sorcery, "transport", "tcp_keepcnt", "5", transport_tcp_keepalive_int_handler, NULL, NULL, 0, 0);
+#if defined(PJ_MAX_SOCKOPT_PARAMS) && PJ_MAX_SOCKOPT_PARAMS >= 5
```

Comment:
I think allowing these to be set will be less confusing. If they aren't allowed then the configuration will be invalid and will state that the option doesn't exist, while documentation will state it does.